cot/src/response/into_response.rs

```diff
@@ -0,0 +1,628 @@
+    where
+        K: TryInto<http::HeaderName, Error = KE>,
+        KE: Into<http::Error>,
+        V: TryInto<http::HeaderValue, Error = VE>,
+        VE: Into<http::Error>,
+        Self: Sized,
+    {
+        let key = key.try_into().map_err(|e| ErrorRepr::from(e.into()))?;
+        let value = value.try_into().map_err(|e| ErrorRepr::from(e.into()))?;
+
+        self.into_response().map(|mut resp| {
+            resp.headers_mut().append(key, value);
+            resp
+        })
+    }
+
+    /// Modifies the response by setting the `Content-Type` header.
+    ///
+    /// # Errors
+    /// Returns an error if the content type value is invalid.
+    fn with_content_type<V, VE>(self, content_type: V) -> cot::Result<Response>
+    where
+        V: TryInto<http::HeaderValue, Error = VE>,
+        VE: Into<http::Error>,
+        Self: Sized,
+    {
+        let content_type = content_type
+            .try_into()
+            .map_err(|e| ErrorRepr::from(e.into()))?;
+
+        self.into_response().map(|mut resp| {
+            resp.headers_mut()
+                .insert(http::header::CONTENT_TYPE, content_type);
+            resp
+        })
+    }
+
+    /// Modifies the response by setting the status code.
+    ///
+    /// # Errors
+    /// Returns an error if the `IntoResponse` conversion fails.
+    fn with_status(self, status: StatusCode) -> cot::Result<Response>
+    where
+        Self: Sized,
+    {
+        self.into_response().map(|mut resp| {
+            *resp.status_mut() = status;
+            resp
+        })
+    }
+
+    /// Modifies the response by setting the body.
+    ///
+    /// # Errors
+    /// Returns an error if the `IntoResponse` conversion fails.
+    fn with_body(self, body: impl Into<Body>) -> cot::Result<Response>
+    where
+        Self: Sized,
+    {
+        self.into_response().map(|mut resp| {
+            *resp.body_mut() = body.into();
+            resp
+        })
+    }
+
+    /// Modifies the response by inserting an extension.
+    ///
+    /// # Errors
+    /// Returns an error if the `IntoResponse` conversion fails.
+    fn with_extension<T>(self, extension: T) -> cot::Result<Response>
+    where
+        T: Clone + Send + Sync + 'static,
+        Self: Sized,
+    {
+        self.into_response().map(|mut resp| {
+            resp.extensions_mut().insert(extension);
+            resp
+        })
+    }
+}
+macro_rules! impl_into_response_for_type_and_mime {
+    ($ty:ty, $mime:expr) => {
+        impl IntoResponse for $ty {
+            fn into_response(self) -> cot::Result<Response> {
+                Body::from(self).with_header(http::header::CONTENT_TYPE, $mime.as_ref())
+            }
+        }
+    };
+}
+
+// General implementations
+
+impl IntoResponse for () {
+    fn into_response(self) -> cot::Result<Response> {
+        Body::empty().into_response()
+    }
+}
+
+impl IntoResponse for Infallible {
+    fn into_response(self) -> cot::Result<Response> {
+        match self {}
+    }
+}
```

Comment:
Since our `IntoResponse for Result<R, E>` doesn't require `E: IntoResponse` now, this shouldn't be needed, right? I don't think there's any reason why anyone would need to return `Infallible` from a request handler.

Reply:
I believe you're right, I've removed it